src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -547,7 +547,7 @@ final void doExecutePreparedStatement(PrepStmtExecCmd command) throws SQLServerE
                 getNextResult();
             }
             catch (SQLException e) {
-                if (retryBasedOnFailedReuseOfCachedHandle(e, attempt))
+                if (retryBasedOnFailedReuseOfCachedHandle(e, attempt) && connection.isStatementPoolingEnabled())
```

Comment:
Suggest changing `retryBasedOnFailedReuseOfCachedHandle` to take care of the `connection.IsStatementPoolingEnabled()` checking, that is part of whether or not to reuse.

Reply:
done.